.github/workflows/build.yml

```diff
@@ -14,13 +14,24 @@ jobs:
       run: |
         echo "ACTIONS_ALLOW_UNSECURE_COMMANDS=true" >> $GITHUB_ENV
 
-    - uses: actions/setup-haskell@v1.1
+    # stack is enabled here to prohibit haskell/actions/setup@v1 from updating cabal. 
+    # Ref: https://github.com/haskell/actions/blob/main/setup/src/setup-haskell.ts#L49
+    - uses: haskell/actions/setup@v1
+      id: setup-haskell
+      name: Setup ghc/cabal (non-alpine)
       with:
-        ghc-version: '8.10'
+        ghc-version: '8.10.5'
+        enable-stack: true 
```

Comment:
[nit] kill this line; we don't use stack

Reply:
I know - but since `spectrometer` build requires `extra-src-file` presents (which we don't include in spectrometer package distribution). Without `enable-stack: true`, build fails as @haskell/actions/setup@v1, will attempt to perform cabal update, leading to CI failure. 

https://github.com/haskell/actions/blob/main/setup/src/setup-haskell.ts#L49. 

I can add this as a comment.